fe_repo/src/App.tsx

```diff
@@ -50,7 +54,7 @@ function App() {
       // handle message sent, update conversation section
       // const response = {text: sendMessage(message), isUser: false}
       // remove sending prompt message
-      sendingMessage.text = analyze(jd);
+      analyze(jd, sendingMessage, refreshMessages);
```

Comment:
Call the `setMessages` inside the `analyze` function, so that the `text` can be a pure string type, and then the component can be separated from the `api.tsx`.